ratis-logservice/src/main/java/org/apache/ratis/logservice/server/LogStateMachine.java

```diff
@@ -230,9 +279,18 @@ public StateMachineStorage getStateMachineStorage() {
 
   }
 
+  private CompletableFuture<Message> recordTime(Timer timer, Task task) {
```

Comment:
Maybe lift this up to some common, abstract StateMachine implementation so we can use it for both the MetadataStateMachine and the LogStateMachine? Or a utility class is fine too.

Reply:
Done